streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java

```diff
@@ -491,175 +585,115 @@ public void shouldThrowExceptionSettingUncaughtExceptionHandlerNotInCreateState(
-        assertNotNull(threadMetadata);
-        assertEquals(2, threadMetadata.size());
-        for (final ThreadMetadata metadata : threadMetadata) {
-            assertTrue("#threadState() was: " + metadata.threadState() + "; expected either RUNNING, STARTING, PARTITIONS_REVOKED, PARTITIONS_ASSIGNED, or CREATED",
-                asList("RUNNING", "STARTING", "PARTITIONS_REVOKED", "PARTITIONS_ASSIGNED", "CREATED").contains(metadata.threadState()));
-            assertEquals(0, metadata.standbyTasks().size());
-            assertEquals(0, metadata.activeTasks().size());
-            final String threadName = metadata.threadName();
-            assertTrue(threadName.startsWith("clientId-StreamThread-"));
-            assertEquals(threadName + "-consumer", metadata.consumerClientId());
-            assertEquals(threadName + "-restore-consumer", metadata.restoreConsumerClientId());
-            assertEquals(Collections.singleton(threadName + "-producer"), metadata.producerClientIds());
-            assertEquals("clientId-admin", metadata.adminClientId());
-        }
-    }
+    public void shouldCleanupOldStateDirs() throws Exception {
+        PowerMock.mockStatic(Executors.class);
+        EasyMock.expect(Executors.newSingleThreadScheduledExecutor(
+            anyObject(ThreadFactory.class)
+        )).andReturn(cleanupSchedule);
+
+        cleanupSchedule.scheduleAtFixedRate(
+            EasyMock.anyObject(Runnable.class),
+            EasyMock.eq(1L),
+            EasyMock.eq(1L),
+            EasyMock.eq(TimeUnit.MILLISECONDS)
+        );
+        EasyMock.expectLastCall().andReturn(null);
+        cleanupSchedule.shutdownNow();
+        EasyMock.expectLastCall().andReturn(null);
+
+        PowerMock.expectNew(StateDirectory.class,
+            anyObject(StreamsConfig.class),
+            anyObject(Time.class),
+            EasyMock.eq(true)
+        ).andReturn(stateDirectory);
+
+        PowerMock.replayAll(Executors.class, cleanupSchedule, stateDirectory);
 
-    @Test
-    public void shouldAllowCleanupBeforeStartAndAfterClose() {
-        try {
-            globalStreams.cleanUp();
-            globalStreams.start();
-        } finally {
-            globalStreams.close();
-        }
-        globalStreams.cleanUp();
-    }
-
-    @Test
-    public void shouldThrowOnCleanupWhileRunning() throws InterruptedException {
-        globalStreams.start();
-        TestUtils.waitForCondition(
-            () -> globalStreams.state() == KafkaStreams.State.RUNNING,
-            "Streams never started.");
-
-        try {
-            globalStreams.cleanUp();
-            fail("Should have thrown IllegalStateException");
-        } catch (final IllegalStateException expected) {
-            assertEquals("Cannot clean up while running.", expected.getMessage());
-        }
-    }
-
-    @Test
-    public void shouldCleanupOldStateDirs() throws InterruptedException {
```

Comment:
These are examples where we should really not rely on checking the underlying state directory files, but just check the mocked state-directory as the expected construction call, etc.